docs/static/azure-module.asciidoc

```diff
@@ -0,0 +1,451 @@
+    var.input.azure_event_hubs.event_hub_connections:
+      - "Endpoint=sb://example1...EntityPath=insights-logs-errors"
+      - "Endpoint=sb://example2...EntityPath=insights-metrics-pt1m"
+-----
+
+===== Advanced configuration sample
+
+Advanced configuration in the `logstash.yml` file supports Event Hub specific options. 
+Advanced configuration is not necessary or recommended for most use cases. Use
+it only if it is required for your deployment scenario. 
+
+You must define the `header` array with `name` in the first position. You can
+define other options in any order. The per Event Hub configuration takes
+precedence. Any values not defined per Event Hub use the global config value.  
+
+In this example 'consumer_group' will be applied to each of the configured Event
+Hubs. Note that 'decorate_events' is defined in both the 'global' and per Event Hub
+configuration. The per Event Hub configuration takes precedence, and the
+global configuration is effectively ignored.
+
+["source","shell",subs="attributes"]
+-----
+modules:
+  - name: azure
+    var.elasticsearch.hosts: "localhost:9200"
+    var.kibana.host: "localhost:5601"
+    var.input.azure_event_hubs.threads: 8
+    var.input.azure_event_hubs.decorate_events: true
+    var.input.azure_event_hubs.consumer_group: logstash
+    var.input.azure_event_hubs.event_hubs:
+      - ["name",                      "event_hub_connection",      "storage_connection",                                      "initial_position", "decorate_events"]
+      - ["insights-operational-logs", "Endpoint=sb://example1...", "DefaultEndpointsProtocol=https;AccountName=example1....", "HEAD",             "true"]
+      - ["insights-metrics-pt1m",     "Endpoint=sb://example2...", "DefaultEndpointsProtocol=https;AccountName=example2....", "TAIL",             "true"]
+      - ["insights-logs-errors",      "Endpoint=sb://example3...", "DefaultEndpointsProtocol=https;AccountName=example3....", "TAIL",             "false"]
+      - ["insights-operational-logs", "Endpoint=sb://example4...", "DefaultEndpointsProtocol=https;AccountName=example4....", "HEAD",             "true"]
+-----
+
+===== Map input options to the logstash.yml file
+
+The configurations in the `logstash.yml` file must be in YAML format. 
+
+Example:
+
+["source","shell",subs="attributes"]
+-----
+var.input.azure_event_hubs.event_hubs:
+  - ["name",                      "event_hub_connection",      "storage_connection",                                      "initial_position", "decorate_events"]
+  - ["insights-operational-logs", "Endpoint=sb://example1...", "DefaultEndpointsProtocol=https;AccountName=example1....", "HEAD",             "true"]
+  - ["insights-metrics-pt1m",     "Endpoint=sb://example2...", "DefaultEndpointsProtocol=https;AccountName=example2....", "TAIL",             "true"]
+-----
+
+[[azure_config_options]]
+===== Configuration options
```

Comment:
Can we add a quick blurb about prefixing the configuration options ? With an example or two?

For example:
threads configuration
Command line:
```
-M "azure.var.input.azure_event_hubs.threads=8"
```
Basic:
```
var.input.azure_event_hubs.threads: 8
```

Advanced:
```
var.input.azure_event_hubs.event_hubs:
  - ["name",                      "event_hub_connection",      "threads"]
  - ["insights-operational-logs", "Endpoint=sb://example1...", 8]
```